src/components/KTable/KTable.vue

```diff
@@ -501,7 +501,15 @@ const props = defineProps({
   },
 })
 
-const emit = defineEmits(['sort', 'ktable-error-cta-clicked', 'ktable-empty-state-cta-clicked', 'row-click', 'cell-click', 'update:table-preferences'])
+const emit = defineEmits<{
+  (e: 'cell-click', value: { data: any }): void
+  (e: 'row-click', value: { data: any }): void
+  (e: 'ktable-error-cta-clicked'): void
+  (e: 'ktable-empty-state-cta-clicked'): void
+  (e: 'update:table-preferences', preferences: TablePreferences): void
+  (e: 'sort', value: { prevKey: string, sortColumnKey: string, sortColumnOrder: string }): void
+  (e: 'state', value: { state: string, hasData: boolean }): void
```

Comment:
suggestion: if the strings state can emit are restricted, can we better type this instead of `string`?